### PR TITLE
fix: stack overflow

### PR DESCRIPTION
**Pull request by smerle33**
correct a parameter for the unit test to go through when checkout from a branch.

### CODE REVIEW

**Review by lemeurherve — APPROVED** [review]

File: vars/infra.groovy [hunk 1]
```diff
@@ -40,7 +40,7 @@ Object checkoutSCM(String repo = null) {
     }
 
     if (env.BRANCH_NAME) {
-        checkout scm
+        checkout repo
```
Comment (timja) [inline vars/infra.groovy:43]: `checkout scm` is the way of doing this, have you got a link for `checkout repo` I've never heard of it

Comment (dduportal) [inline vars/infra.groovy:43]:
```suggestion
        checkout scm
```

**Review by dduportal — CHANGES_REQUESTED** [review]
+ 1 with Tim remark.

**Comment by smerle33** [comment]
@timja what do you think of removing the checkout method that is generating the stackoverflow from within the unit test ? I think that it should not have the same name as "checkout" ....

File: test/groovy/InfraStepTests.groovy [hunk 2]
```diff
@@ -58,7 +58,7 @@ class InfraStepTests extends BaseTest {
   }
 
   @Test
-  @Ignore("Some stackoverflow issues")
+  //@Ignore("Some stackoverflow issues")
```
Comment (dduportal) [inline test/groovy/InfraStepTests.groovy:61]:
```suggestion
```

**Comment by timja** [comment]
> @timja what do you think of removing the checkout method that is generating the stackoverflow from within the unit test ? I think that it should not have the same name as "checkout" ....

I think okay, seems to just be a legacy method delegating to checkoutSCM

**Review by dduportal — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Pull request merged**